pkg/api/v1/alert_configurations.go

```diff
@@ -208,6 +228,31 @@ type Notification struct {
 	Roles []string `json:"roles,omitempty"`
 }
 
+func (in Notification) Key() string {
+	return in.APITokenRef.Key() + "|" +
+		in.ChannelName + "|" +
+		in.DatadogAPIKeyRef.Key() + "|" +
+		in.DatadogRegion + "|" +
+		strconv.Itoa(admin.GetOrDefault(in.DelayMin, 0)) + "|" +
+		in.EmailAddress + "|" +
+		strconv.FormatBool(admin.GetOrDefault(in.EmailEnabled, false)) + "|" +
+		in.FlowdockAPITokenRef.Key() + "|" +
+		in.FlowName + "|" +
+		strconv.Itoa(in.IntervalMin) + "|" +
+		in.MobileNumber + "|" +
+		in.OpsGenieAPIKeyRef.Key() + "|" +
+		in.OpsGenieRegion + "|" +
+		in.OrgName + "|" +
+		in.ServiceKeyRef.Key() + "|" +
+		strconv.FormatBool(admin.GetOrDefault(in.SMSEnabled, false)) + "|" +
+		in.TeamID + "|" +
+		in.TeamName + "|" +
+		in.TypeName + "|" +
+		in.Username + "|" +
+		in.VictorOpsSecretRef.Key() + "|" +
+		"[" + strings.Join(in.Roles, ",") + "]"
```

Comment:
Why not use a hash code for all the strings instead? does it need to be readable? I to not think will ever be (readable)

Reply:
hashes are not sortable which is the point of this function

Reply:
This is just an example of implementation. It has no impact on the current logic

Reply:
yes, we need sortable keys for normalization

Reply:
oh, you indeed have a point about being stable. as long as that invariant is maintained, then yes, stable hashes are good enough